src/Storages/Kafka/StorageKafka2.h

```diff
@@ -0,0 +1,247 @@
+#pragma once
+
+#include <Core/BackgroundSchedulePool.h>
+#include <Core/Block.h>
+#include <Core/Types.h>
+#include <Storages/IStorage.h>
+#include <Storages/Kafka/KafkaConsumer2.h>
+#include <Storages/Kafka/KafkaSettings.h>
+#include <Common/Macros.h>
+#include <Common/SettingsChanges.h>
+#include <Common/ThreadStatus.h>
+#include <Common/ZooKeeper/ZooKeeper.h>
+
+#include <Poco/Semaphore.h>
+
+#include <atomic>
+#include <filesystem>
+#include <list>
+#include <mutex>
+#include <rdkafka.h>
+
+namespace cppkafka
+{
+
+class Configuration;
+
+}
+
+namespace DB
+{
+
+template <typename TStorageKafka>
+struct StorageKafkaInterceptors;
+
+using KafkaConsumer2Ptr = std::shared_ptr<KafkaConsumer2>;
+
+/** Implements a Kafka queue table engine that can be used as a persistent queue / buffer,
+  * or as a basic building block for creating pipelines with a continuous insertion / ETL.
```

Comment:
Makes sense to expand here on some implementation details. Just briefly. Like how it uses keeper, what data it stores there.